### ISSUE TITLE
[coc.nvim]: Error on "jumpDefinition": definition provider not found for current buffer, your language server doesn't support it.

### ISSUE THREAD

**Issue by dprophete** [iamcco/coc-flutter#140]
Hi guys,

Not sure if it's related to upgrading to the latest flutter + dart, but coc-flutter now always reports:
`[coc.nvim]: Error on "jumpDefinition": definition provider not found for current buffer, your language server doesn't support it.` when I try to jump to definition or get the references.

Note that coc itself seems to work properly (coc goto definition works with elm files for instance)

And note that the other parts of `coc-flutter` do work properly (I can run `:CocList FlutterEmulators`, etc...)

Everything worked fine until about a week ago. Again, not sure if it's a change in coc, in coc-flutter, or it's related to moving to the latest flutter. Here is the output of `flutter --version`:
```
Flutter 2.5.0 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 4cc385b4b8 (11 days ago) • 2021-09-07 23:01:49 -0700
Engine • revision f0826da7ef
Tools • Dart 2.14.0
```

Here are the coc extensions I have installed (result of `CocUpdate`):
```
Update finished

- ✓ coc-css Current version 1.2.6 is up to date.
- ✓ coc-flutter Current version 1.9.5 is up to date.
- ✓ coc-highlight Current version 1.2.8 is up to date.
- ✓ coc-html Current version 1.4.1 is up to date.
- ✓ coc-json Current version 1.3.6 is up to date.
- ✓ coc-kotlin Current version 0.0.7 is up to date.
- ✓ coc-lists Current version 1.4.2 is up to date.
- ✓ coc-prettier Current version 1.1.24 is up to date.
- ✓ coc-tsserver Current version 1.8.6 is up to date.
```

Also, here is the output of `CocList services` 
```
    languageserver.elmLS [init] elm
    highlight [init]
  * flutter [running] dart
```

The dart lsp server seems to be running properly:
```
>> ps aux| grep dart
didier           63520   0.4  1.5  8846876 505420   ??  Ss   Fri06PM  15:19.78 /Users/didier/Library/Developer/CoreSimulator/Devices/2D658C4A-81FA-405B-989B-54E15140AC09/data/Containers/Bundle/Application/4F67E123-2181-4031-85A4-E1298013F74F/Runner.app/Runner --enable-dart-profiling --enable-checked-mode --verify-entry-points --observatory-port=0
didier           61986   0.3  0.5  5433028 158180   ??  S     8:24PM   0:03.08 /Users/didier/tmp/flutter/bin/cache/dart-sdk/bin/dart --disable-dart-dev --packages=/Users/didier/tmp/flutter/packages/flutter_tools/.packages /Users/didier/tmp/flutter/bin/cache/flutter_tools.snapshot daemon
didier           62265   0.0  0.0  4408672    804 s010  S+    8:26PM   0:00.00 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn dart
didier           61968   0.0  0.8  5664504 252604   ??  S     8:24PM   0:04.77 /Users/didier/tmp/flutter/bin/cache/dart-sdk/bin/dart /Users/didier/tmp/flutter/bin/cache/dart-sdk/bin/snapshots/analysis_server.dart.snapshot --lsp
```

I am using the latest version of coc with
```
Plug 'neoclide/coc.nvim', {'branch': 'release'}
```

Regarding vim+dart, I tried with and without `vim-lsc` and `vim-lsc-dart`. That didn't solve the problem.
```
Plug 'dart-lang/dart-vim-plugin'
"Plug 'natebosch/vim-lsc'
"Plug 'natebosch/vim-lsc-dart'
"let g:lsc_auto_map = v:true
```

I am at a loss... Because of this I had to go back to using intellij for flutter development :-<

Hope somebody can get me out of my misery here :-) Thx guys.

**Comment by llucax** [comment]
Hi, I'm having a similar issue. I've been away from Flutter for around 4 months. Now I tried to work again on a project and also noticed I barely have any of the benefits of coc-flutter. I don't get analyzer indications, nor go to definition, nor auto-import or even completion from classes in the imports I already imported. I still get too the `flutter` service working and using `CocCommand` or `CocList` involving flutter all work, and also other "intellisense" completion work too (like methods in classes).

Another example of a basic feature not working is pressing `K` to get info on a symbol, I get:

```
Error on request (doHover): Vim(return):Error invoking 'doHover' on channel 4 (coc):
hover provider not found for current buffer, your language server doesn't support it.
```

To see if the issue is some weird interaction between some modules, I even started a new `vimrc` with just this:

```
call plug#begin(stdpath('data') . '/vim-plug-test-flutter')"
Plug 'dart-lang/dart-vim-plugin' " Dart programming language
Plug 'neoclide/coc.nvim', {'branch': 'release'} " CoC intellisense engine
call plug#end()
```

And then ran `CocInstall coc-flutter`. Still I'm getting this very reduced experience. coc is working perfectly fine with all other languages I use (python, rust, shell, JSON, etc.).

Any ideas on what could be going wrong and how to fix it is extremely appreciated, as this is making developing for flutter very old-school :)

**Comment by Kavantix** [comment]
Just released v1.9.6 which should fix these issues

**Comment by dprophete** [comment]
Very nice @Kavantix . Happy to report that it does indeed fix the issue. Thx.

**Comment by llucax** [comment]
Me too, thanks!

**Comment by llucax** [comment]
One question, since we are here and I don't know of a better place to ask. As part of my coming back to Flutter after a while being away, I updated to use the new flutter lints rules. I noticed I don't get the errors/warnings/info I get via `flutter analyze` inside vim with coc-flutter. Is this normal/expected? Is there a way to enable getting analyze issues as info/warn/errors in vim? I couldn't find any obvious options in this project's readme.
